charts/spire/charts/spire-agent/values.yaml

```
@@ -166,6 +166,37 @@ nodeAttestor:
   k8sPsat:
     ## @param nodeAttestor.k8sPsat.enabled Enable Psat k8s Node Attestor
     enabled: true
+  tpmDirect:
```

Maybe just name this. Trying to understand what Direct means in this context.

```suggestion
  tpm:
```

---

Marked as Direct to keep room for a second feature in the future. There is a second tpm mode that spire supports here: https://github.com/spiffe/spire/blob/main/doc/plugin_agent_nodeattestor_tpm_devid.md

Direct uses the tpm's private key hash directly for identification. This is the simplest way to use a tpm.

DeviceID requires an extra server and an out of band registration mechanism. But is the only way to attest some devices (Apple I think)

---

ok, thanks for explaining, makes sense to keep room for extension. Is `direct` a uniformal way of referring to this tpm approach? If so I'm fine. Just wanter to pick your brain on the naming convention.

---

I don't know of any standard naming around it. Also open to different names. Though naming things is hard.

---

Yup agree. I'm not very into tpm stuff so just asked my 101 questions here on the PR. I guess it is good to go. @faisal-memon feel free to approve and merge if you don't see any other hurdles.

---

Naming looks ok to me. Keeping room for the second tpm plugin makes sense